### PR TITLE
Ally Events

### PR DESCRIPTION
**Pull request by ceeedric**
Added ally events

#### Description: 

Adds the following events:
 - NationAcceptAllyRequestEvent
 - NationDenyAllyRequestEvent
 - NationPreAcceptAllyRequestEvent (cancellable)
 - NationPreDenyAllyRequestEvent (cancellable)
 - NationRemoveAllyEvent
 - NationPreRemoveAllyEvent (cancellable)
____
#### New Nodes/Commands/ConfigOptions: 

____
#### Relevant Towny Issue ticket:

____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

File: src/com/palmergames/bukkit/towny/command/NationCommand.java [hunk 1]
```diff
@@ -9,17 +9,7 @@
...
-import com.palmergames.bukkit.towny.event.NationPreAddEnemyEvent;
-import com.palmergames.bukkit.towny.event.NationPreRemoveEnemyEvent;
-import com.palmergames.bukkit.towny.event.NationRemoveEnemyEvent;
-import com.palmergames.bukkit.towny.event.NationRequestAllyNationEvent;
-import com.palmergames.bukkit.towny.event.NewNationEvent;
-import com.palmergames.bukkit.towny.event.NationPreTransactionEvent;
-import com.palmergames.bukkit.towny.event.NationTransactionEvent;
-import com.palmergames.bukkit.towny.event.NationPreAddTownEvent;
-import com.palmergames.bukkit.towny.event.NationPreRenameEvent;
+import com.palmergames.bukkit.towny.event.*;
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/NationCommand.java:12]: We don't do the wildcard imports. Please return it to how it was before.

File: src/com/palmergames/bukkit/towny/command/NationCommand.java [hunk 2]
```diff
@@ -2006,19 +2007,30 @@ public void nationAlly(Resident resident, final Nation nation, List<Nation> alli
...
 								nation.addAlly(targetNation);
+								
+								
 							} catch (AlreadyRegisteredException e) {
 								e.printStackTrace();
 							}
+							
 							TownyMessaging.sendPrefixedNationMessage(nation, Translation.of("msg_allied_nations", resident.getName(), targetNation.getName()));
 							TownyMessaging.sendPrefixedNationMessage(targetNation, Translation.of("msg_added_ally", nation.getName()));
+
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/NationCommand.java:2018]: Remove whitespacing changes.

**Review by LlmDl — CHANGES_REQUESTED** [review]
Aside from the minor requested changes I'd like to know if the cancel-able events are actually removing the invitations that are denied. If they are not being removed (I suspect they aren't) then the king would end up just trying accept it over and over. 

There also needs to be a cancelled message shown to the person who's had their accept/deny action denied. You only need to add the new language string to the english.yml and I will copy it to the others.

**Review by silverwolfg11 — CHANGES_REQUESTED** [review]
I don't really see the need for dividing each of these events up into a `pre` event and the actual event. The reason being is that this division is usually made when there is a case where we do not want the event to be cancelled and do not want to present an option of the event being cancelled.

In the cases of the events here, all of them are fine if they are cancelled and there is no case where we want to prevent cancellation of them.

Thus, I think it's best just to remove all the `pre` events and make the actual events cancellable. Developers who want to listen to the final state of these events can just listen to the event on `MONITOR` priority.

**Review by LlmDl — APPROVED** [review]

**Pull request merged**